code/modules/mob/mob_grab_specials.dm

```diff
@@ -86,6 +86,53 @@
 
 	attack.handle_eye_attack(attacker, target)
 
+/obj/item/grab/proc/dropkick(mob/living/carbon/target, mob/living/carbon/human/attacker)
+	if(state < GRAB_AGGRESSIVE) //blue grab check
+		to_chat(attacker, SPAN_WARNING("You require a better grab to do this."))
+		return
+	if(target.lying)
+		return
+	visible_message(SPAN_DANGER("[attacker] dropkicks [target], pushing \him onward!"))
+	attacker.Weaken(2)
+	target.Weaken(6) //the target will fly over tables, railings, etc.
+	var/kick_dir = get_dir(attacker, target)
+	if(attacker.loc == target.loc) // if we are on the same tile(e.g. neck grab), turn the direction to still push them away
+		kick_dir = turn(kick_dir, 180)
+	target.throw_at(get_edge_target_turf(target, kick_dir), 3, 1)
+	//deal damage AFTER the kick
+	var/damage = (attacker.stats.getStat(STAT_ROB) / 3)
+	target.damage_through_armor(damage, BRUTE, BP_GROIN, ARMOR_MELEE)
+	//admin messaging
+	attacker.attack_log += text("\[[time_stamp()]\] <font color='red'>Dropkicked [target.name] ([target.ckey])</font>")
+	target.attack_log += text("\[[time_stamp()]\] <font color='orange'>Dropkicked by [attacker.name] ([attacker.ckey])</font>")	
+	msg_admin_attack("[key_name(attacker)] has dropkicked [key_name(target)]")	
+	//kill the grab
+	attacker.drop_from_inventory(src)
```

Comment:
It works that way? No need to kill the grab before throwing the target?

Reply:
it does, yea
~~I state for the third time that I test my code~~

Reply:
wait what does this question mean

Reply:
If you tested it it's likely fine. I was asking if the fact that you kill the grab after throwing the person away does not block the throw. The game being like "Hey this dude is being `throw_at` but since he is being grabbed that not possible so the function does nothing".

Reply:
Oh. Yeah it works. I had to ask kirov about fixing some stuff because it did not work at first